x-pack/plugins/infra/common/runtime_types.ts

```diff
@@ -23,3 +24,16 @@ export const decodeOrThrow = <A, O, I>(
   createError: ErrorFactory = createPlainError
 ) => (inputValue: I) =>
   pipe(runtimeType.decode(inputValue), fold(throwErrors(createError), identity));
+
+type ValdidationResult<Value> = ReturnType<RouteValidationFunction<Value>>;
+
+export const createValidationFunction = <A, O, I>(
+  runtimeType: Type<A, O, I>
```

Comment:
Small nit: I personally find generics easier / quicker to parse with meaningful type variables, could we change these from `A, O, I`?

Reply:
Good point, I absolutely agree. I stuck with the io-ts naming scheme during experimentation and forgot to clean it up afterwards.